generateLocalesPatterns/index.js

```diff
@@ -0,0 +1,48 @@
+const fs = require("fs");
+const locales = require("./localLocales");
+
+const obj = {};
+
+const dataOptions = (data, code) => {
+  if (!data.type) return new Error;
+
+  const valueOptions = {
+    month: data.value.length == 1 ? "M" : "MM",
+    day: data.value.length == 1 ? "D" : "DD",
+    year: "YYYY",
+    literal: data.value,
+    default: () => {
+      return console.error(`${data.type} and  ${data.value} is not a data type - code: ${code}`);
+    }
+  };
+
+  return valueOptions[data.type] || valueOptions.default();
+}
+
+const parsePhoneFormat = (code) => {
+  const formatObj = new Intl.DateTimeFormat(code).formatToParts(new Date())
+  return formatObj.map((dateItem) => dataOptions(dateItem, code)).join("").replace(",", "")
+};
+
+const getEra = (code) => {
+  const formatObj = new Intl.DateTimeFormat(code).formatToParts(new Date());
+  const era = formatObj.filter(item => item.type === 'era')
+  return !!era.length
+};
+
+for (const key in locales) {
+  if (locales.hasOwnProperty(key)) {
+    const code = locales[key].code;
+    obj[key] = {
+      name: locales[key].name,
+      code: locales[key].code,
+      dateFormat: parsePhoneFormat(code),
+      era: getEra(code),
+    };
+  }
+}
+
+var json = JSON.stringify(obj);
+fs.writeFile("locales.json", json, "utf8", function (err) {
+  err ? new Error(err) : console.log("Success");
+});
```

Comment:
```js
const showError = () => err ? new Error(err) : console.log("Success");
fs.writeFile("locales.json", json, "utf8", showError);
```

Talvez essa função possa ser revista, entendi que ela loga erros mas talvez tenha um jeito da gente tratar isso melhor? De qualquer forma sei que não é prioritário, só uma nota aí

Reply:
então vou pensar em algo melhor para a função de erro quando for adicionar o parse de telefone de cada Região